src/CoreWCF.Primitives/src/CoreWCF/IdentityModel/Tokens/WSSecurityXXX2005.cs

```diff
@@ -0,0 +1,249 @@
+﻿using CoreWCF.IdentityModel.Selectors;
+using CoreWCF.Security.Tokens;
+using System;
+using System.Collections.Generic;
+using System.Xml;
+using static CoreWCF.IdentityModel.Selectors.SecurityTokenSerializer;
+
+namespace CoreWCF.IdentityModel.Tokens
+{
+    internal class WSSecurityXXX2005 : WSSecurityJan2004
+    {
+        public WSSecurityXXX2005(KeyInfoSerializer securityTokenSerializer)
+            : base(securityTokenSerializer)
+        {
+        }
+
+        public override void PopulateStrEntries(IList<StrEntry> strEntries)
+        {
+            PopulateJan2004StrEntries(strEntries);
+         //   strEntries.Add(new SamlDirectStrEntry());
+          //  strEntries.Add(new X509ThumbprintStrEntry(this.SecurityTokenSerializer.EmitBspRequiredAttributes));
+            strEntries.Add(new EncryptedKeyHashStrEntry(this.SecurityTokenSerializer.EmitBspRequiredAttributes));
```

Comment:
X509ThumbprintStrEntry is probably going to be needed as the plan is to support username profile and X509 certificate profile.

Reply:
ported ..